vizro-core/docs/pages/user-guides/data.md

```diff
@@ -1,199 +1,39 @@
-
-If the data you are visualizing is a [`kedro_datasets.pandas`](https://docs.kedro.
-org/en/stable/kedro_datasets.html) type from a Kedro data catalog, you can use
-Vizro's [Kedro integration](integration.md#kedro) to connect your charts to [Kedro's Data Catalog](https://docs.kedro.org/en/stable/data/index.html).
-
-If it is not a `kedro_datasets.pandas` type, you need to build a
-data connector to load the data from the data catalog and convert it to a Pandas
-DataFrame, before you can register it with Vizro Data Manager.
+Vizro supports two different types of data:
+
+* [Static data](static-data.md): pandas DataFrame. This is the simplest method and best to use if you do not need the more advanced functionality of dynamic data.
+* [Dynamic data](dynamic-data.md): function that returns a pandas DataFrame. This is a bit more complex to understand but has more advanced functionality such as the ability to refresh data while the dashboard is running.
+
+The following flowchart shows what you need to consider when choosing how to set up your data.
+``` mermaid
+graph TD
+  refresh["`Do you need your data to refresh while the dashboard is running?`"]
+  specification["`Do you need to specify your dashboard through a configuration language like YAML?`"]
+  dynamic([Use dynamic data referenced by name])
+  static-direct([Use static data supplied directly])
+  static-name([Use static data referenced by name])
+
+  refresh -- No --> specification
+  refresh -- Yes --> dynamic
+  specification -- No --> static-direct
+  specification -- Yes --> static-name
+  
+  click static-direct href "../static-data#supply-directly"
+  click static-name href "../static-data#reference-by-name"
+  click dynamic href "../dynamic-data"
+```
+
+??? note "Static vs. dynamic data comparison"
+
+    This table gives a full comparison between static and dynamic data. Do not worry if you do not yet understand everything in it; it will become clearer after reading more about [static data](static-data.md) and [dynamic data](dynamic-data.md)!
+
+    |                                                               | Static           | Dynamic                                  |
+    |---------------------------------------------------------------|------------------|------------------------------------------|
+    | Required Python type                                          | pandas DataFrame | Function that returns a pandas DataFrame |
+    | Can be supplied directly in `data_frame` argument of `figure` | Yes              | No                                       |
+    | Can be referenced by name after adding to Data Manager        | Yes              | Yes                                      |
+    | Can be refreshed while dashboard is running                   | No               | Yes                                      |
+    | Production-ready                                              | Yes              | Yes                                      |
+
```

Comment:
These are great questions that we discussed and have a good understanding of. Just for the record let me put here my very rough diagram of what happens with static data if you do this:
```
df = px.data.iris()
page.components = [
    vm.Graph(id=1, figure=px.scatter(df, ...)),
    vm.Graph(id=2, figure=px.scatter(df, ...))
]
```
![image](https://github.com/mckinsey/vizro/assets/49395058/8f2928f6-0bb9-4c7f-9702-ffdcb795f573)

There will be two `_StaticDataset` instances but they both point to the same `df` in memory. This is copied when you do `data_manager[name].load()`.

As per the comment I added to the `load` method this feels like the correct way round (rather than copying when you do `_StaticDataset.__init__`) because it's analogous to this:
```
data_manager["iris"] = px.data.iris()
page.components = [
    vm.Graph(id=1, figure=px.scatter("iris", ...)), 
    vm.Graph(id=2, figure=px.scatter("iris", ...))
]
```
Here there's still just one underlying dataframe and the only difference is that there's one `_StaticDataset`.

So to answer all the questions:

>  - Is there any time/memory usage difference between (and is the difference even more obvious if the app is running on multiple processes):
>     - when two graphs use static data directly and
>     - when two graphs use static data by name referencing?

These are essentially the same. The only overhead with using static data directly is extra `_StaticDataset` instances but these take up basically no memory/time.

> - What happens for the time/memory usage if we use  `lambda: static_data` instead? (and is the difference even more obvious if the app is running on multiple processes)?

This is also basically the same. There's still just one `df` in memory. In this case the `df` _doesn't_ get copied on load. In theory this is a bit risky because you could mutate `df`, but this is ok because we don't want to do a `copy` operation (that can be expensive) unnecessarily given that dynamic data should come from a pure function and not consuming a globally scoped variable. So basically don't do this and mutate `df`, just like you shouldn't with the static data case we discussed this morning...
```
from vizro.managers._data_manager import data_manager
import pandas as pd
df = pd.DataFrame()
data_manager["x"] = lambda: df
data_manager["y"] = lambda: df

assert data_manager["x"].load() is data_manager["y"].load() is df
```

> - What happens if the underlying cache is:
>   - SimpleCache,
>   - FileSystemCache,
>   - RedisCache 
>   - (and is the difference even more obvious if the app is running on multiple processes)?

This is only relevant at all for dynamic data, and in that case the basic answer is "not sure, we'd need to do some proper profiling". My guess though is:
* time: file > redis > simple
* memory: simple > redis > file
... and of course simple doesn't work for multiple processes.

> Do we want to give more context here on how static data (supplied direct and by name) and dynamic data (null, simple, file_system, redis) work behind the scene? Then our users can see the difference in time/memory usage between these options and can wisely decide which option suits them best?

I decided not to put this in the public docs since I think it's too technical/irrelevant for most people, but we can add some more guidance on choice between redis vs. filesystem cache and how the cache works in future (hopefully when we understand the comparison better ourselves).